The Weather App Project Submission

MussieTeka opened the pull request:
This pull request adds the Weather App, a React-based web application for viewing weather information for multiple cities. It utilizes the OpenWeatherMap API to fetch weather data and provides features such as adding new cities, filtering cities by temperature, and displaying detailed weather information for individual cities.

## Changes Made
- Implemented the Weather App components: Home, AddCity, CityList, Filter, and Details.
- Integrated the OpenWeatherMap API to fetch weather data for multiple cities and individual cities.
- Added functionality to add new cities, filter cities by temperature, and view detailed weather information.
- Styled the components using CSS to create an appealing user interface.
- Created a detailed and professional README.md file with installation instructions, usage guide, and contribution guidelines.
- I have added testing to ensure the correctness and functionality of the Weather App.

## Checklist
- [x] Tested the Weather App locally and verified that it functions as expected.
- [x] Reviewed the code for readability, maintainability, and adherence to best practices.
- [x] Updated the README.md file with clear instructions and information about the project.

## Video Presentation

Please find my presentation [here](https://www.loom.com/share/83cca757c09240fbbbfe58753ff18afd).

Thank you for reviewing my work!

TadesseDev reviewed (changes requested):
### Hi @MussieTeka 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7](https://user-images.githubusercontent.com/69077061/212385281-4797c58a-d358-423a-aa7c-832c22203865.gif)




## Highlights
 - [x] Good job overall following the template design ✅
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
- [ ] Dear @MussieTeka, I like your presentation, and it's almost complete. the one thing missing is the specific code that you are proud of. Please be reminded I have watched you have walked us through the code, but we need to see a specific code you find interesting or you are proud of. 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

MussieTeka commented on README.md:
```diff
@@ -1,2 +1,204 @@
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- [React](https://reactjs.org/)
+- [Create React App](https://create-react-app.dev/)
+- [GitHub](https://github.com/)
+- [Microverse](https://www.microverse.org/)
+- [OpenWeatherMap API](https://openweathermap.org/api): External API for fetching weather data
+
```
Thank you for your feedback @TadesseDev . I have updated the acknowledgment section of our readme file to appropriately credit Nelson Sakwa, the author of the original design, as required by the Creative Commons License.

MussieTeka commented on README.md:
```diff
@@ -1,2 +1,204 @@
+- View detailed weather information for individual cities
+- Responsive design for mobile and desktop devices
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a id="live-demo"></a>
+
+- Click [here](https://the-weather-app-kakb.onrender.com/) to view the live Demo. 
```
Noted. I have fixed the issue by adding "s" to "http" to make the connection secure. As a result, all functionalities now work correctly on the live version.

MussieTeka commented on src/components/Home.js:
```diff
@@ -0,0 +1,68 @@
+    };
+
+    fetchWeatherData();
+  }, [dispatch]);
```
Thank you @TadesseDev for the suggestion to use middleware like Redux Thunks for enabling async logic in Redux calls. Although I have decided not to implement this feature currently due to project scope and time constraints, I appreciate your input. I will consider adding it in the future.

MussieTeka commented on src/App.js:
```diff
@@ -0,0 +1,19 @@
+const App = () => (
+  <Provider store={store}>
+    <Router>
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/details/:id" element={<Details />} />
+      </Routes>
+    </Router>
```
Thank you for your feedback @TadesseDev  on the design. I have made the necessary updates to align with the original design guidelines. Changes include adjusting typography, layout, and composition to match the desired specifications. I appreciate your input in helping me improve the design implementation.

MussieTeka commented:
Dear @TadesseDev,

I have updated my presentation based on your feedback, fixed the functionality issues on the live version, appropriately credited the author as required, and made adjustments to the design to align it with the original guidelines.

Best,
Mussie

OLIPLICHE commented on src/components/CityList.js:
```diff
@@ -0,0 +1,36 @@
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import '../style/CityList.css';
+
+const CityList = ({ cities, getWeatherImage }) => (
+  <div className="city-list-container">
+    {cities.map((city) => (
+      <div key={city.id} className="city-box">
+        <img src={getWeatherImage(city.main.temp)} alt="Weather" className="weather-image" />
+        <Link to={`/details/${city.id}`} className="city-link">
+          {city.name}
+        </Link>
+        <p className="city-temperature">
+          {city.main.temp}
+          °C
+        </p>
+      </div>
+    ))}
+  </div>
+);
```
- Hi, you did a nice job working on the design as requested by the previous reviewer, but you forgot to implement the arrows on the top right of each city. Without these arrows, it will be difficult for the user to Enter the city map to check the details.
- Your App:
![Capture](https://github.com/MussieTeka/the-weather-app/assets/63972670/9c9cf1a0-3be2-46ea-ad14-f541fc4026a0)
- Original:


![Capture](https://github.com/MussieTeka/the-weather-app/assets/63972670/a24401cb-477e-45da-a912-ab9e9bd48c9e)

MussieTeka replied:
Thank you for your feedback @OLIPLICHE. I have now added and positioned an arrow on the top right corner of each city card.

OLIPLICHE reviewed (changes requested):
Hi @MussieTeka, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Nice video presentation 👍 
- Have implemented most of the changes requested by the previous review 👍 
- The App is working fine 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

anagudelogu reviewed (approved):
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @MussieTeka, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 

<img src="https://media0.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
